bitcoin-magazine/bitcoin-explained/bitcoin-core-24.0.md

```diff
@@ -0,0 +1,1150 @@
+
+Sjors Provoost: 00:11:35
+
+And so it'd be nice to get rid of those things entirely because, you know, they are confusing and, you know, we probably don't want to add new ones.
+So In order to get rid of them, you still need to fix this resource exhaustion attack that we just talked about.
+So then how do you do that?
+Well, the trick is to download headers twice, as I said.
+So what you do is you download them once, and when you're downloading them once, you don't save them.
+You just look at them, check if they're correct, and you throw them away.
+This means it does not use any of your disk space.
+Then if you see enough work at the end, if you checked all the headers and you see the proof of work is enough, it's the longest chain, Then you ask the peer, hey, can you send them again, please?
+And you download them again.
+
+
+Aaron van Wirdum: 00:12:21
+
+I think I mentioned yesterday when we went over this in our preparation, this is kind of how my attention span usually works.
+
+Sjors Provoost: 00:12:28
+
+Exactly.
+You first let me ramble on a bit, and then you think, this might actually be interesting.
+And then you say, what did you say again?
+Please repeat.
+
+Aaron van Wirdum: 00:12:35
+
+Can you repeat that for me?
+
+Aaron van Wirdum: 00:12:37
+
+Yeah, so this is how Bitcoin nodes now actually work.
+
+Sjors Provoost: 00:12:39
+
+That's right.
+
+Aaron van Wirdum: 00:12:40
+
+They first listen, they don't store anything, but then if it sounds actually interesting, if they see the proof of work, they just ask again and they get all the blocks.
+
+Sjors Provoost: 00:12:47
+
+They're like, wow, that's cool.
+
+Aaron van Wirdum: 00:12:48
+
+But they get all the headers again, I guess.
+Or do they now also get the blocks?
+
+Sjors Provoost: 00:12:51
+
+Well, then once those headers have been downloaded the second time, things continue as always.
+You just start fetching blocks for the headers.
+Now there's, of course, a little gotcha there because, well, who says that the second time you're getting the same headers as the first time?
+You don't know that, because you didn't store them.
+So what you do instead is you store one checksum, which is a one-bit checksum, every 50,000 blocks or so.
+So every 50,000 headers you store a zero or a one depending on the contents of those headers and you do that every 50,000 headers.
+That is very little information but it turns out that it's actually quite difficult for anybody to fake that information.
+So it's for anybody to create fake headers that do match your checksum, even though it's a very small checksum.
+
+## checksums
+
+Aaron van Wirdum: 00:13:36
+
+Right, yeah, a checksum is essentially you add up all kinds of numbers and you get a very short number which doesn't in itself prove anything, but I guess if you do it a bunch of times then.
+
+Sjors Provoost: 00:13:46
+
+If you add up the same numbers you're gonna get the same checksum.
+
+Aaron van Wirdum: 00:13:48
+
+Right, exactly.
```

Comment:
This kind of lines don't contribute to the transcript's content - you can just delete this and continue with the same speaker. This transcript has a lot back n forth between participants, so you'll notice the presence of numerous filler words or non-essential lines that need removal.